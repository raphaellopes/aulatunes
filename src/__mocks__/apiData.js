export const albums = [
  {
    'im:name': {
      label: 'Love & Hate',
    },
    'im:image': [
      {
        label: 'https://is5-ssl.mzstatic.com/image/thumb/Music124/v4/dc/1f/cb/dc1fcbe6-7ff3-7064-e0da-4ce65ee89efb/00602547859044.rgb.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is2-ssl.mzstatic.com/image/thumb/Music124/v4/dc/1f/cb/dc1fcbe6-7ff3-7064-e0da-4ce65ee89efb/00602547859044.rgb.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/dc/1f/cb/dc1fcbe6-7ff3-7064-e0da-4ce65ee89efb/00602547859044.rgb.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$8.99',
      attributes: {
        amount: '8.99000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/love-hate/1440832200?uo=2',
      attributes: {
        'im:id': '1440832200',
      },
    },
    'im:artist': {
      label: 'Michael Kiwanuka',
      attributes: {
        href: 'https://music.apple.com/us/artist/michael-kiwanuka/433457346?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '15',
        term: 'R&B/Soul',
        scheme: 'https://music.apple.com/us/genre/music-r-b-soul/id15?uo=2',
        label: 'R&B/Soul',
      },
    },
  },
  {
    'im:name': {
      label: 'Cypress Grove',
    },
    'im:image': [
      {
        label: 'https://is1-ssl.mzstatic.com/image/thumb/Music123/v4/6d/61/20/6d6120c2-4eb7-8d86-14c9-5a29d26057f8/850007715281.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is3-ssl.mzstatic.com/image/thumb/Music123/v4/6d/61/20/6d6120c2-4eb7-8d86-14c9-5a29d26057f8/850007715281.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is4-ssl.mzstatic.com/image/thumb/Music123/v4/6d/61/20/6d6120c2-4eb7-8d86-14c9-5a29d26057f8/850007715281.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$9.99',
      attributes: {
        amount: '9.99000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/cypress-grove/1474602159?uo=2',
      attributes: {
        'im:id': '1474602159',
      },
    },
    'im:artist': {
      label: 'Jimmy "Duck" Holmes',
      attributes: {
        href: 'https://music.apple.com/us/artist/jimmy-duck-holmes/265026843?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '2',
        term: 'Blues',
        scheme: 'https://music.apple.com/us/genre/music-blues/id2?uo=2',
        label: 'Blues',
      },
    },
  },
];

export const songs = [
  {
    'im:name': {
      label: 'Leave The Door Open',
    },
    'im:image': [
      {
        label: 'https://is5-ssl.mzstatic.com/image/thumb/Music114/v4/36/8b/5c/368b5c12-725e-e85d-36be-5287461cc0a2/075679788337.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is4-ssl.mzstatic.com/image/thumb/Music114/v4/36/8b/5c/368b5c12-725e-e85d-36be-5287461cc0a2/075679788337.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is4-ssl.mzstatic.com/image/thumb/Music114/v4/36/8b/5c/368b5c12-725e-e85d-36be-5287461cc0a2/075679788337.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$1.29',
      attributes: {
        amount: '1.29000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/leave-the-door-open/1556096966?i=1556097162&uo=2',
      attributes: {
        'im:id': '1556097162',
      },
    },
    'im:artist': {
      label: 'Bruno Mars, Anderson .Paak & Silk Sonic',
      attributes: {
        href: 'https://music.apple.com/us/artist/bruno-mars/278873078?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '15',
        term: 'R&B/Soul',
        scheme: 'https://music.apple.com/us/genre/music-r-b-soul/id15?uo=2',
        label: 'R&B/Soul',
      },
    },
  },
  {
    'im:name': {
      label: "Don't Start Now",
    },
    'im:image': [
      {
        label: 'https://is1-ssl.mzstatic.com/image/thumb/Music124/v4/fc/aa/c9/fcaac9e1-327d-3e61-56f2-8911bc63c9d1/190295322175.jpg/55x55bb.png',
        attributes: {
          height: '55',
        },
      },
      {
        label: 'https://is5-ssl.mzstatic.com/image/thumb/Music124/v4/fc/aa/c9/fcaac9e1-327d-3e61-56f2-8911bc63c9d1/190295322175.jpg/60x60bb.png',
        attributes: {
          height: '60',
        },
      },
      {
        label: 'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/fc/aa/c9/fcaac9e1-327d-3e61-56f2-8911bc63c9d1/190295322175.jpg/170x170bb.png',
        attributes: {
          height: '170',
        },
      },
    ],
    'im:price': {
      label: '$0.69',
      attributes: {
        amount: '0.69000',
        currency: 'USD',
      },
    },
    id: {
      label: 'https://music.apple.com/us/album/dont-start-now/1484636581?i=1484636829&uo=2',
      attributes: {
        'im:id': '1484636829',
      },
    },
    'im:artist': {
      label: 'Dua Lipa',
      attributes: {
        href: 'https://music.apple.com/us/artist/dua-lipa/1031397873?uo=2',
      },
    },
    category: {
      attributes: {
        'im:id': '14',
        term: 'Pop',
        scheme: 'https://music.apple.com/us/genre/music-pop/id14?uo=2',
        label: 'Pop',
      },
    },
  },
];
