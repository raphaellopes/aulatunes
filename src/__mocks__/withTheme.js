import React from 'react';
import { ThemeProvider } from 'styled-components';

export const withTheme = (Component) => (
  <ThemeProvider theme={{ mode: 'light' }}>
    {Component}
  </ThemeProvider>
);
