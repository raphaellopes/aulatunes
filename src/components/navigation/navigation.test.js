import { render, screen, fireEvent } from '@testing-library/react';

import { withTheme } from '../../__mocks__/withTheme';
import { Navigation } from './index';

describe('components | navigation', () => {
  const options = [
    { label: 'Some option 1', value: 'some-option-1' },
    { label: 'Some option-2', value: 'some-option-2' },
  ];

  test('should render the Navigation', () => {
    const { container } = render(withTheme(
      <Navigation
        options={options}
        onClickOption={jest.fn}
        active="some-option-1"
      />,
    ));
    expect(container).toMatchSnapshot();
  });

  test('should dispatch the onClickOption when prop is set', () => {
    const onClickOption = jest.fn();
    render(
      <Navigation
        options={options}
        onClickOption={onClickOption}
        active="some-option-1"
      />,
    );
    fireEvent.click(screen.getByText(/Some option-2/i));
    expect(onClickOption).toHaveBeenCalledWith('some-option-2');
  });
});
