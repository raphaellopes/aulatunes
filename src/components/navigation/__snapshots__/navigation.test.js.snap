// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | navigation should render the Navigation 1`] = `
.c1 {
  background-color: transparent;
  padding: 10px;
  border-radius: 20px;
  text-transform: uppercase;
  font-weight: 600;
  font-size: 14px;
  border: 1px solid #791abe;
  background-color: #791abe;
  color: #ffffff;
}

.c3 {
  background-color: transparent;
  padding: 10px;
  border-radius: 20px;
  text-transform: uppercase;
  font-weight: 600;
  font-size: 14px;
  cursor: pointer;
  border: 1px solid #d6d6d7;
  color: #31405c;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px 0;
}

.c2:not(:last-child) {
  margin-right: 10px;
}

<div>
  <nav
    class="c0"
  >
    <button
      class="c1 c2"
      data-testid="menu-option-some-option-1-active"
    >
      Some option 1
    </button>
    <button
      class="c3 c2"
      data-testid="menu-option-some-option-2"
    >
      Some option-2
    </button>
  </nav>
</div>
`;
